packages/block-library/src/paragraph/edit.js

```diff
@@ -124,90 +136,65 @@ function ParagraphBlock( {
 					}
 				/>
 			</BlockControls>
-			<InspectorControls>
-				<PanelBody title={ __( 'Text settings' ) }>
-					<FontSizePicker
-						value={ fontSize.size }
-						onChange={ setFontSize }
-					/>
-					<ToggleControl
-						label={ __( 'Drop Cap' ) }
-						checked={ !! dropCap }
-						onChange={ () =>
-							setAttributes( { dropCap: ! dropCap } )
-						}
-						help={
-							dropCap
-								? __( 'Showing large initial letter.' )
-								: __( 'Toggle to show a large initial letter.' )
-						}
-					/>
-				</PanelBody>
-			</InspectorControls>
+			{ InspectorControlsFontSizePanel }
 			{ InspectorControlsColorPanel }
-			<BackgroundColor>
-				<TextColor>
-					<RichText
-						ref={ ref }
-						identifier="content"
-						tagName="p"
-						className={ classnames(
-							'wp-block-paragraph',
-							className,
-							{
-								'has-drop-cap': dropCap,
-								[ `has-text-align-${ align }` ]: align,
-								[ fontSize.class ]: fontSize.class,
-							}
-						) }
-						style={ {
-							fontSize: fontSize.size
-								? fontSize.size + 'px'
-								: undefined,
-							direction,
-							minHeight: dropCapMinimumHeight,
-						} }
-						value={ content }
-						onChange={ ( newContent ) =>
-							setAttributes( { content: newContent } )
-						}
-						onSplit={ ( value ) => {
-							if ( ! value ) {
-								return createBlock( name );
+			<FontSize>
+				<BackgroundColor>
+					<TextColor>
```

Comment:
Do you think we should find a solution for this? what if we keep adding hooks and nesting,

Reply:
I don't think it will ever scale to be an issue.

Reply:
For me personally,  it already feels like an issue. Granted that it's not a huge one.